### PR TITLE
Migrate Channel.RemoveAllDeactivatedMembers to Sync by default

### PR DESCRIPTION
**Pull request by gupsho**
#### Summary
Migrate "Channel.RemoveAllDeactivatedMembers" to Sync by default

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/11250

### CODE REVIEW

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -1694,9 +1694,8 @@ func (s SqlChannelStore) RemoveMember(channelId string, userId string) store.Sto
 	})
 }
 
-func (s SqlChannelStore) RemoveAllDeactivatedMembers(channelId string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		query := `
+func (s SqlChannelStore) RemoveAllDeactivatedMembers(channelId string) *model.AppError {
+	query := `
 			DELETE
```
Comment (jespino) [inline store/sqlstore/channel_store.go]: Do you mind to unindent the query string too?

Comment (gupsho) [reply store/sqlstore/channel_store.go]: Done.

**Comment by gupsho** [comment]
> A small change required, otherwise looks good to me. Thanks @gupsho! 👍

Fixed.

**Comment by jespino** [comment]
Merged! Thanks @gupsho! :dancing_men:

**Pull request merged**